### PR TITLE
airport challenge

### PR DESCRIPTION
**Pull request by giusepped**

### CODE REVIEW

File: lib/airport.rb [hunk 1]
```diff
@@ -1,2 +1,46 @@
...
+    plane.land
+    @landed_planes << plane
+    plane
+  end
+
+  def traffic_control
+    fail 'Airport is full!' if full?
+  end
+
+  def weather_control
```
Comment (dwatson62) [inline lib/airport.rb]: Can use rand(5) for random weather generating
Or if you are worried about the hound, can do
weather = [stormy]
4.times { weather.push('sunny')

File: lib/airport.rb [hunk 2]
```diff
@@ -1,2 +1,46 @@
...
+
+  def let_take_off plane
+    fail 'Stormy weather!' if stormy?
+    plane.take_off
+    @landed_planes.delete plane
+  end
+
+  def let_land plane
+    fail 'Stormy weather!' if stormy?
+    traffic_control
```
Comment (dwatson62) [inline lib/airport.rb]: Good for separating traffic_control into a new method

File: lib/plane.rb [hunk 3]
```diff
@@ -1,2 +1,23 @@
 class Plane
+
+  def initialize
+    @flying = true
+  end
+
+  def flying?
```
Comment (dwatson62) [inline lib/plane.rb]: Can use attr_reader for flying

File: spec/airport_spec.rb [hunk 4]
```diff
@@ -12,22 +12,50 @@
 # If the airport is full then no planes can land
 
 describe Airport do
+  let(:plane) { double(:plane) }
+
+  it 'has a default capacity' do
+    expect(subject.capacity).to eq Airport::DEFAULT_CAPACITY
```
Comment (dwatson62) [inline spec/airport_spec.rb]: Can specify what the default capacity is in the test eg.
it 'has a default capacity of 50'
expect(subject.capacity).to eq 50

File: spec/airport_spec.rb [hunk 5]
```diff
@@ -12,22 +12,50 @@
...
+      allow(subject).to receive(:weather_control) { 'sunny' }
+      allow(plane).to receive(:land)
+      expect(subject.let_land plane).to be plane
+    end
   end
 
   describe 'traffic control' do
     context 'when airport is full' do
-      xit 'does not allow a plane to land'
+      it 'does not allow a plane to land' do
```
Comment (dwatson62) [inline spec/airport_spec.rb]: Can use a before(:each) block to refactor
          allow(subject).to receive(:weather_control) { 'sunny' }
          allow(plane).to receive(:land)
before(:each) do

end

File: spec/airport_spec.rb [hunk 6]
```diff
@@ -40,9 +68,21 @@
     # the plane can not land, and must not be in the airport
 
     context 'when weather conditions are stormy' do
```
Comment (dwatson62) [inline spec/airport_spec.rb]: Same as down here, before(:each)

File: Gemfile.lock [hunk 7]
```diff
@@ -59,3 +59,6 @@ DEPENDENCIES
   capybara
```
Comment (dwatson62) [inline Gemfile.lock]: Don't forget to write a readme! Include what the program does, and what you wanted it do or not had time to finish. Also add installation instructions and how to run the test.

File: lib/airport.rb [hunk 8]
```diff
@@ -1,2 +1,46 @@
...
+    plane
+  end
+
+  def traffic_control
+    fail 'Airport is full!' if full?
+  end
+
+  def weather_control
+    ['stormy', '', '', '', '', '', '', ''].sample
+  end
```
Comment (danldb) [inline lib/airport.rb]: Use a symbol instead of a string - more efficient and immutable.

File: lib/plane.rb [hunk 9]
```diff
@@ -1,2 +1,23 @@
...
+  end
+
+  def land
+    @flying ? @flying = false : (fail 'Plane has already landed')
+  end
+
+  def take_off
+    !@flying ? @flying = true : (fail 'Plane is already flying')
+  end
+
```
Comment (danldb) [inline lib/plane.rb]: why not use landed? instead of !flying?

**Comment by danldb** [comment]
:checkered_flag: 

Well done Giuseppe!

**Pull request closed without merging**